packages/dds/tree/src/test/simple-tree/api/schemaCreationUtilities.spec.ts

```diff
@@ -126,35 +205,45 @@ describe("schemaCreationUtilities", () => {
 	});
 
 	it("enum value switch", () => {
-		const Mode = enumFromStrings(schema, ["Fun", "Cool", "Bonus"]);
+		const Mode = enumFromStrings(schema, ["Fun", "Bonus"]);
 		class Parent extends schema.object("Parent", { mode: Mode.schema }) {}
-		const config = new TreeViewConfiguration({ schema: Parent });
+		const root = new Parent({ mode: new Mode.Bonus() });
+		const mode = root.mode;
 
-		const factory = new TreeFactory({});
-		const tree = factory.create(
-			new MockFluidDataStoreRuntime({ idCompressor: testIdCompressor }),
-			"tree",
-		);
-		const view: TreeView<typeof Parent> = tree.viewWith(config);
-		view.initialize(
-			new Parent({
-				mode: new Mode.Bonus(),
-			}),
-		);
-		const mode = view.root.mode;
-		switch (mode.value) {
+		// mode.value must be copied out into its own variable for the switch to narrow it correctly.
+		const value = mode.value;
+
+		switch (value) {
 			case "Fun": {
 				assert.fail();
 			}
-			case "Cool": {
+			case "Bonus": {
+				// This one runs
+				break;
+			}
+			default:
+				unreachableCase(value);
+		}
+	});
+
+	it("enum value switch - minimal", () => {
+		const Mode = enumFromStrings(schema, ["Fun", "Bonus"]);
+		type Mode = TreeNodeFromImplicitAllowedTypes<typeof Mode.schema>;
+		const node = new Mode.Bonus() as Mode;
```

Comment:
Does `const node: Mode = new Mode.Bonus()` not compile?

Reply:
It does not since typescript narrows it. I added a comment about this on one of the places this workaround is done.